gen/plugin.go

```diff
@@ -44,13 +44,17 @@ type generateServiceBuilder struct {
 	// ThriftFile -> Service name -> Service ID
 	serviceIDs map[string]map[serviceName]api.ServiceID
 
+	// To ensure there are no duplicates
+	rootModules map[api.ModuleID]struct{}
+
 	// To ensure there are no duplicates
 	rootServices map[api.ServiceID]struct{}
 }
 
 func newGenerateServiceBuilder(i thriftPackageImporter) *generateServiceBuilder {
 	return &generateServiceBuilder{
 		GenerateServiceRequest: api.GenerateServiceRequest{
+			RootModules:   make([]api.ModuleID, 0),
```

Comment:
we should give it a default capacity of 1 since that's what'll usually go in there.

```suggestion
			RootModules:   make([]api.ModuleID, 0, 1),
```